lib/web_ui/lib/src/engine/view_embedder/flutter_view_manager.dart

```diff
@@ -96,12 +107,14 @@ class FlutterViewManager {
     return _jsViewOptions[viewId];
   }
 
-  /// Returns the [viewId] if [rootElement] corresponds to any of the [views].
-  int? viewIdForRootElement(DomElement rootElement)  {
-    for(final EngineFlutterView view in views) {
-      if (view.dom.rootElement == rootElement) {
-        return view.viewId;
+  EngineFlutterView? findViewForElement(DomElement? element) {
+    DomElement? current = element;
+    while (current != null) {
+      final EngineFlutterView? view = _elementToView[current];
```

Comment:
Why this `while` and not `closest`? Is iterating our own cache faster than a brower `querySelector`/`closest`?

Reply:
I don't like the idea of depending on tag name and html attributes. But I'm not feeling strongly about it, so I'll switch to `closest` since it might be faster.

Reply:
Done